### PR TITLE
Add Proxy Upgradable Contract Example w HTS

### LINKED ISSUE #84: Add proxy upgrade contract example w HTS
### Problem

Hedera is able to support upgradeable contracts using the proxy method.
However, no example exists in this repo yet

### Solution

Add 
- a set of contracts example under `contracts/proxy-upgrade` that showcase how an upgradeable contract may be utilized w HTS logic e.g. very simple DEX ref.
- unit tests that show valid contract execution
- [Stretch] manual upgrade flow in a test that verifies eventual updated logic

Note there may be some dev tool limitations due to CREATE 1 logic by Hedera vs Ethereum but this should either have documentation or a work around provided.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by georgi-l95**
**Description**:

Add two Proxy Upgradable UUPS Contracts and Tests:
- Counter
- Simple DEX with Hedera token

Bump Network and Mirror Node images.
Bump NodeJS version to the latest LTS - 18.12.1

**Related issue(s)**:

Fixes #84 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: contracts/proxy-upgrade/exchange/Exchange.sol [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import "../../hts-precompile/IHederaTokenService.sol";
+import "../../hts-precompile/HederaResponseCodes.sol";
+
+contract Exchange is OwnableUpgradeable, UUPSUpgradeable {
+    address public tokenAddress;
+    address private precompile;
```
Comment (natanasow) [inline contracts/proxy-upgrade/exchange/Exchange.sol:13]: Let's assign the value directly here and make it constant `address constant private precompile = address(0x167);`

**Review by natanasow — APPROVED** [review]
LG

File: contracts/proxy-upgrade/exchange/Exchange.sol (hunk 1 above)
Comment (georgi-l95) [reply contracts/proxy-upgrade/exchange/Exchange.sol]: Done.

**Review by natanasow — DISMISSED** [review]

**Comment by lukelee-sl** [comment]
LG.  Are there any "negative" tests we want to consider?  For example, trying to call at the version() function before the upgrade.  I think it's fine without but wanted to just raise the question.

**Review by lukelee-sl — DISMISSED** [review]

**Comment by georgi-l95** [comment]
You're right, i've added negative cases.

**Review by lukelee-sl — APPROVED** [review]

**Pull request merged**